### DISCUSSION TITLE
Question: Actions and adding additional data.

### DISCUSSION THREAD

**Discussion by wh1337** [hasura/graphql-engine#7052] [Question]
I was wondering, is there a way with the actions, to add additional data to a request?

For example, if a user creates a new `post`, I would like to add a new entry into the `posts` table, but also add a new entry into the `posts_history` table. Would the best way to accomplish this be to have it call an external API via the actions and then have that update and add a new entry into `posts_history`?

**Comment by MarcusBondezan** [comment]
Hey @cryptic-ai I think in this case you could use the [Hasura Events](https://hasura.io/docs/latest/graphql/core/event-triggers/index.html) feature. So, whenever a row is CREATED in yout `post` table, Hasura triggers an event for you to add that new row in the `post_history` table. Of course, the event will be triggering a function developed by you with all the customizations you need.
